internal/lsp/commands/parse_test.go

```diff
@@ -0,0 +1,110 @@
+package commands
+
+import (
+	"testing"
+
+	"github.com/open-policy-agent/opa/ast"
+
+	"github.com/styrainc/regal/internal/lsp/types"
+)
+
+func TestParse(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		ExecuteCommandParams types.ExecuteCommandParams
+		ParseOptions         ParseOptions
+		ExpectedTarget       string
+		ExpectedLocation     *ast.Location
+	}{
+		"extract target only": {
+			ExecuteCommandParams: types.ExecuteCommandParams{
+				Command:   "example",
+				Arguments: []interface{}{"target"},
+			},
+			ParseOptions:     ParseOptions{TargetArgIndex: 0},
+			ExpectedTarget:   "target",
+			ExpectedLocation: nil,
+		},
+		"extract target and location": {
+			ExecuteCommandParams: types.ExecuteCommandParams{
+				Command:   "example",
+				Arguments: []interface{}{"target", "1", 2}, // different types for testing, but should be strings
```

Comment:
I suppose we could have standardized on args just being a single object for all commands, which we could serialize into a struct. But this might be more aligned with what the LSP "expects".

Reply:
Yeah I wasn't so sure about this but figured that we'd go with this until we ran into some issues. It's all internal so seemed ok to go with this for now.